.github/workflows/ansible-release.yml

```diff
@@ -0,0 +1,144 @@
+        with:
+          python-version: '3.11'
+
+      - name: Install dependencies
+        working-directory: antsibull
+        run: |
+          python3 -m pip install ansible-core
+          python3 -m pip install antsibull
+          ansible-galaxy collection install community.general
+
+      - name: Checking out to a new branch
+        working-directory: antsibull/build/ansible-build-data
+        run: |
+          git checkout -b "publish-${ANSIBLE_VERSION}"
+
+      - name: Setting the user details
+        run: |
+          git config --global user.name "Github Actions"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+      # Run the playbook according to the current release process
+
+      - name: Building a release with the defaults
+        working-directory: antsibull
+        run: >-
+          ansible-playbook -vv playbooks/build-single-release.yaml
+          -e antsibull_data_reset=false
+          -e "antsibull_ansible_version=${ANSIBLE_VERSION}"
+
+        env:
+          # Make result better readable
+          ANSIBLE_CALLBACK_RESULT_FORMAT: yaml
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: sdist-and-wheel
+          path: antsibull/build/ansible-*.*
+
+      - name: Commit ansible-build-data and push the changes to github
+        working-directory: 'antsibull/build/ansible-build-data/${{ inputs.ansible-major-version }}'
+        run: |
+         git add *.rst *.build *.deps *.yaml *.in validate-tags-ignores
+         git commit -m "${{ env.CI_COMMIT_MESSAGE }}"
+         git push origin "publish-${ANSIBLE_VERSION}"
+
+      - name: Create PR to the ansible-build-data
+        working-directory: antsibull/build/ansible-build-data
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: >-
+          gh pr create
+          --base main
+          --head "publish-${{ inputs.ansible-version }}"
+          --title "${{ env.CI_COMMIT_MESSAGE }}"
+          --body "${{ env.CI_COMMIT_MESSAGE }}"
+
+  # publish job downloads the arifacts and publish it to PyPI
+
+  publish:
+    needs: [build]
+    name: 'Upload Ansible (${{ inputs.ansible-version }}) to PyPI'
+    runs-on: ubuntu-latest
+    environment:
+      name: pypirelease
+      url: https://pypi.org/project/ansible
+    permissions:
+      id-token: write
+
+    steps:
+
+    - name: Download artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: sdist-and-wheel
+        path: dist/
+
+    - name: Upload Ansible sdist and wheel to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+
+  # git-tag job creates the git tag
+
+  git-tag:
+    needs: publish
+    name: 'Creates git tag for Ansible (${{ inputs.ansible-version }})'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+
+      - name: Check out ansible-build-data
+        uses: actions/checkout@v3
```

Comment:
I think, this needs an explicit `ref` input set. By default, it'll probably target the commit attached to the original `workflow_dispatch` event that triggered this workflow run. However, we need to tag the merge commit after the PR, not the commit before the PR, since we want the tagged version to contain changes from the PR, right?

Reply:
If you want to postpone this — it's fine. But then, you need to at least force this specific checkout step to fetch the top of the main branch, not some old pre-PR commit:
```suggestion
        uses: actions/checkout@v3
        with:
          ref: ${{ github.event.repository.default_branch }}
```
(and implement what I suggested above in a follow-up PR)

Reply:
> To implement this, we need to save the PR number as an output of the `build` job above. Then, we'd access that PR number in this job, and query GitHub API (or maybe just use GH CLI, not sure) to retrieve the PR metadata and see which commit exactly is the merge commit associated with that PR.


I will file a separate PR on this context as a future enhancement.